pkg/cloud/ibmcloud/actuators/machine/actuator.go

```diff
@@ -231,24 +231,20 @@ func (ic *IBMCloudClient) Update(ctx context.Context, cluster *clusterv1.Cluster
 	}
 
 	if guest.Domain == nil ||
-		guest.MaxMemory == nil ||
 		guest.OperatingSystemReferenceCode == nil ||
-		guest.LocalDiskFlag == nil ||
+		(guest.SupplementalCreateObjectOptions != nil && guest.SupplementalCreateObjectOptions.FlavorKeyName == nil) ||
```

Comment:
This logic seems still have problem. When only `guest.SupplementalCreateObjectOptions == nil`, it will not return error and continue to run. However, later access to `*guest.SupplementalCreateObjectOptions.FlavorKeyName ` will crash.

Reply:
now let's assume guest.SupplementalCreateObjectOptions must not be nil otherwise the Flavor is not set, so let's check guest.SupplementalCreateObjectOptions and guest.SupplementalCreateObjectOptions.Flavor both not null here..